drivers/builtin/include/mbedtls/config_adjust_legacy_crypto.h

```diff
@@ -378,4 +378,15 @@
 #define MBEDTLS_SSL_HAVE_AEAD
 #endif
 
+/* Deduce MBEDTLS_CTR_DRBG_USE_128_BIT_KEY from MBEDTLS_PSA_CRYPTO_RNG_STRENGTH */
+#if defined(MBEDTLS_PSA_CRYPTO_RNG_STRENGTH)
+#if (MBEDTLS_PSA_CRYPTO_RNG_STRENGTH == 128)
+#define MBEDTLS_AES_ONLY_128_BIT_KEY
```

Comment:
No, we can't do that: AES can be used for things other than the RNG! The most obvious case for having 256-bit AES but a 128-bit RNG strength is if you want to have the fastest RNG that the library has to offer (CTR_DRBG with AES-128), but you still want to offer AES-256 for applications that insist on it.

`MBEDTLS_AES_ONLY_128_BIT_KEY` is incompatible with a 256-bit RNG strength _if the RNG is CTR\_DRBG_, but not if the RNG is something else. Checking that would be the job of `check_config.h`, not here.

What we need to set here is not `MBEDTLS_AES_ONLY_128_BIT_KEY` but `MBEDTLS_CTR_DRBG_USE_128_BIT_KEY` (and the logic here is correct for that).